src/passes/param-utils.cpp

```diff
@@ -79,11 +79,8 @@ bool removeParameter(const std::vector<Function*>& funcs,
     bool hasUnremovable =
       EffectAnalyzer(runner->options, *module, operands[index])
         .hasUnremovableSideEffects();
-    bool wouldChangeType =
-      call->type == Type::unreachable && !call->isReturn &&
-      std::any_of(operands.begin(), operands.end(), [](Expression* operand) {
-        return operand->type == Type::unreachable;
-      });
+    bool wouldChangeType = call->type == Type::unreachable && !call->isReturn &&
+                           operands[index]->type == Type::unreachable;
```

Comment:
We could be even more precise and additionally check that none of the _other_ operands is unreachable. If one of them is, then removing this one still won't change the type. Of course if this DAE is run after DCE, that won't matter, so maybe it's not worth doing.

Reply:
Yes, I think that could help but is not likely to in the general case. For now let's not add that extra complexity.